### DISCUSSION TITLE
Extend API rules and filters via event hooks

### DISCUSSION THREAD

**Discussion by diogomartino** [pocketbase/pocketbase#3325] [Q&A] [answered]
So, I need a very specific and kinda complex update rule on one of my collections so it's not really feasible to use the UI to create it in one line. So, is there any way to handle complex cases like this in a custom hook? I thought that adding logic to the `onRecordBeforeCreate` might do the trick, but I didn't find any way of getting the id of the user that's doing the request (same as `@request.auth.id`) and that's needed for my use case.

**Comment by ganigeorgiev** [comment, marked as answer]
See https://github.com/pocketbase/pocketbase/discussions/3258. The equivalent for the JS hooks would be something like:

```js
onRecordBeforeCreateRequest((e) => {
    const info = $apis.requestInfo(e.httpContext)
    console.log(info.admin?.id)      // will be undefined if not authenticated as admin
    console.log(info.authRecord?.id) // will be undefined if not authenticated as auth record
}, "yourCollection")
```
_(https://pocketbase.io/docs/js-routing/#retrieving-the-current-auth-state; `e.httpContext` is `c` in the examples )_

**Reply by diogomartino** [reply]
Thank you, don't know how I missed that.
